File: array/926.py
class Solution:
    def minFlipsMonoIncr(self, S):
        """
        :type S: str
        :rtype: int
        """
        count0 = S.count('0')
        count1 = 0
        ret = len(S) - count0 
        for i in range(len(S)):
            if S[i] == '0':
                count0 -= 1
            else:
                ret = min(ret, count0+count1)
                count1 += 1
        return ret

        

File: dp/926.cc
class Solution {
public:
    int minFlipsMonoIncr(string s) {
        int counter_one = 0, counter_flip = 0;
        for (const auto& c : s) {
            if (c == '1') {
                ++counter_one;    
            } else {
                counter_flip = std::min(counter_one, counter_flip+1);
            }
        }
        return counter_flip;
    }
};
